localstack/services/lambda_/event_source_listeners/adapters.py

```diff
@@ -207,6 +208,10 @@ def _get_role_for_function(self, function_arn: str) -> str:
         function_name, qualifier, account, region = function_locators_from_arn(function_arn)
         store = lambda_stores[account][region]
         function = store.functions.get(function_name)
+
+        if not function:
+            raise FunctionNotFoundError(f"function not found: {function_arn}")
```

Comment:
Thanks @thrau 🙏 

that exception was already helpful:

`2024-01-04T19:24:52.166 DEBUG --- [functhread30] l.s.l.e.sqs_event_source_l : 'NoneType' object has no attribute 'latest'`

turned into

`2024-01-04T20:07:45.347 DEBUG --- [functhread33] l.s.l.e.sqs_event_source_l : function not found: arn:aws:lambda:us-east-1:000000000000:function:test-lambda-perf-77809b32`

=> It looks like SQS event source listener does not always properly shut down because I am getting this error after a successful test run of [test_lambda_event_source_mapping_sqs](https://github.com/localstack/localstack/pull/9972/files#diff-c626c358a191e4cc10999a6ed5f301e2f5a7eca9d379e91731268810803e3a8cR345)